### PR TITLE
[TC] Support any kind of environment in Env. Registry

### PR DESCRIPTION
**Pull request by IrinaComindaru**
Related issue https://github.com/capsulajs/environment-registry/issues/16

### CODE REVIEW

File: tests/features/register.feature [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+Scenario: Calling register method registers the environment of the provided envKey and env
+  When  User calls register method the <envKey> and <env>
+    |<envKey>  | <env>      |
+    |'develop' | environment|
+    |'develop' | ' '        |
+  Then  Registration of the environment is performed with success
+  And   Subscribing to environments method returns the registered environment
+
+Scenario: Calling register method with an envKey already registered
+  Given An environment with envKey 'develop' is registered in Environment Registry
```
Comment (MikeSquall) [inline tests/features/register.feature:23]:
```suggestion
  Given An environment with envKey 'develop' is registered in Environment Registry
  And   The environment 'env' property is `undefined`
```

File: tests/features/register.feature [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+  When  User calls register method with valid env and envKey 'develop'
```
Comment (MikeSquall) [inline tests/features/register.feature]:
```suggestion
  When  User calls register method with envKey 'develop'
  And   with env '123'
```

File: tests/features/register.feature [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+  Then  The value of the existing env is updated with the new one
+  And   Subscribing to environments method returns the updated environment
```
Comment (MikeSquall) [inline tests/features/register.feature]:
```suggestion
  And   Subscribing to environments method returns 123
```

File: tests/features/register.feature [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+         |undefined      |
+         |123            |
+         |[]             |
+         |['test']       |
+         |{}             |
+         |{ test: 'test'}|
+  Then  The validation error 'envKeyIsNotCorrect' is returned
+
+
+ //For future implementation
```
Comment (MikeSquall) [inline tests/features/register.feature]:
```suggestion

```

**Comment by MikeSquall** [comment]
I changed something in the issue. Instead of having `env: any` in the interface, we have now `env: T` where `T` is a type you can pass to the interface.

So can you:
- add a test that check if I instantiate an `EnvRegistry<T>`, I can't register an `env` with type `Z`
- add some details to the 1st scenario 
?

File: tests/features/register.feature (hunk 4 above)
Comment (stephanebenayoun) [reply tests/features/register.feature]: To remove

File: tests/features/register.feature [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+  Given Environment Registry with register method
+  And   a valid 'envKey' is a non empty string
+  And   an 'env' can be anything
+
+
+Scenario: Calling register method registers the environment of the provided envKey and env
+  Given An environment registry of <type>
+  When  User calls register method with the <envKey> and <env>
+    |<envKey>  | <env>          | <type>     |
+    |'develop' | undefined      | any        |
```
Comment (dmitriyslaym) [inline tests/features/register.feature]: type "undefined"

File: tests/features/register.feature [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+    |'develop' | null           | any        |
```
Comment (dmitriyslaym) [inline tests/features/register.feature]: technically speaking - type "object" )

Comment (dmitriyslaym) [reply tests/features/register.feature]: @MikeSquall Do we really want to allow even falsy values to be allowed as env?

**Review by dmitriyslaym — APPROVED** [review]

**Pull request merged**